Scripts/gamedata.gd

```diff
@@ -1065,3 +1132,47 @@ func remove_entity_from_map(map_id: String, entity_type: String, entity_id: Stri
 	print_debug("Entity removal operations completed for all levels.")
 	var map_data_json = JSON.stringify(mapdata.duplicate(), "\t")
 	Helper.json_helper.write_json_file(fileToLoad, map_data_json)
+
+# A skill is being deleted from the data
+# We have to remove it from everything that references it
+func on_skill_deleted(skill_id: String):
+	var changes_made = { "value": false }  # Using a Dictionary to hold the change status
+	var skill_data = get_data_by_id(Gamedata.data.skills, skill_id)
+
+	if skill_data.is_empty():
+		print_debug("Skill with ID", skill_id, "not found.")
+		return
+
+	# This callable will remove the skill references from items that reference this skill.
+	var myfunc: Callable = func (item_id):
```

Comment:
Rename to something meaningful. For example `fn_remove_references_from_items`. Even better if you move it to a separate `func remove_references_from_items` function, instead of using callable like that.

Reply:
Renamed it